### ISSUE TITLE
"Download" buttons for binaries should trigger client-side "Save As..." dialog, rather than loading in browser

### ISSUE THREAD

**Issue by GeoffFroh** [densho/ddr-public#27]
Buttons for downloading binary fullsize and access files are currently simple direct links to the content. When the user clicks a download button, the browser attempts to display the content directly.

The desired behavior is that clicking the download button triggers the browser's "Save As..." dialog.

Notes:

HTML5 supports this functionality through the "download" attribute on links (http://www.w3.org/TR/html5/links.html#downloading-resources); however, this is not currently viable with just over 50% browser compatibility (http://caniuse.com/#search=download)

The common server-side implementation is to set relevant file mimetypes to application/octet-stream. Often this is done through a script that is triggered, rather than setting the mimetype directly in the server config

**Comment by gjost** [comment]
It's a little more complicated than the examples because we have two levels of Nginx here (`tank` and `tulie`).
With the following configs this URL results in the "Save As..." dialog:
http://ddr.densho.org/download/media/ddr-densho-67/ddr-densho-67-113-mezzanine-59a42802c5-a.jpg
The original URL (minus the `/download` part) lets you view the image normally:
http://ddr.densho.org/media/ddr-densho-67/ddr-densho-67-113-mezzanine-59a42802c5-a.jpg

NOTE: For some reason this works for JPEGs and TIFFs but not for PDFs.

tank.conf:
```
server {
    listen      80;
    server_name ddr.densho.org;
    ...
    # Force "Save As..."
    location /download/ {
        #expires 30d;
        if ($request_filename ~ "^.*/(.+\.jpg|jpeg|pdf|tif|tiff)$"){
            set $fname $1;
            add_header Content-Disposition 'attachment; filename="$fname"';
        }
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header Host $http_host;
        proxy_redirect off;
        proxy_pass http://ddr_media;
        break;
    }
    ...
```

tulie.conf:
```
server {
    listen       80;
    server_name  default;
    ...
    location /download/ {
        alias  /var/www/;
    }
    location / {
        root  /var/www;
    }
}
```

**Comment by gjost** [comment]
Partially addressed in 9ec76f9c3d.

**Comment by gjost** [comment]
Fixed as of ba04ef2.  Also requires `public.download_url` setting in `ddrlocal/debian/conf/ddr.cfg`.

* ba04ef2  file download URLs use DOWNLOAD_URL, trigger browser "Save As..." (geoffrey jost)
* 577c110  S3-bucket-style paths for access, orig files (geoffrey jost)
* 97831d9  function for getting Entity signature File object (geoffrey jost)
* 74b50dd  add MEDIA_URL, STATIC_URL, and DOWNLOAD_URL to app-wide context (geoffrey jost)
* 9ec76f9  DOWNLOAD_URL for "Download full size"; brings up "Save As..." (geoffrey
